presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -214,8 +214,7 @@ void PrestoServer::run() {
 
   initializeCoordinatorDiscoverer();
   std::unique_ptr<Announcer> announcer;
-  if (coordinatorDiscoverer_ != nullptr &&
-      !coordinatorDiscoverer_->updateAddress().empty()) {
+  if (coordinatorDiscoverer_ != nullptr) {
     announcer = std::make_unique<Announcer>(
```

Comment:
Shall we let Presto server to own Announcer as well?

Reply:
Yeah we can do that as a follow up. Shall we keep this PR as a fix PR then I can have another PR refactor the announcer ownership thingy.